packages/uui-text-copy/lib/uui-text-copy.element.ts

```diff
@@ -0,0 +1,171 @@
+import { css, html, LitElement } from 'lit';
+import { property } from 'lit/decorators.js';
+import { defineElement } from '@umbraco-ui/uui-base/lib/registration';
+import { demandCustomElement } from '@umbraco-ui/uui-base/lib/utils';
+import { UUIButtonElement } from '@umbraco-ui/uui-button/lib';
+import { UUIInterfaceColor, UUIInterfaceLook } from '@umbraco-ui/uui-base';
+import { LabelMixin } from '@umbraco-ui/uui-base/lib/mixins';
+import { UUITextCopyEvent } from './UUITextCopyEvent';
+
+/**
+ * @summary A button to trigger text content to be copied to the clipboard
+ * @element uui-text-copy
+ * @dependency uui-button
+ * @dependency uui-icon
+ * @fires {UUITextCopyEvent} copying - Fires before the content is about to copied to the clipboard and can be used to transform or modify the data before its added to the clipboard
+ * @fires {UUITextCopyEvent} copied - Fires when the content is copied to the clipboard
+ * @slot - Use to replace the default content of 'Copy' and the copy icon
+ */
+@defineElement('uui-text-copy')
```

Comment:
Could we call this `uui-string-copy-button` and then it should be extending uui-button so we do not need to property drill so much. :-)

Reply:
Just to summen up, based on the conversation above, this component should and can inherit from `<uui-button>` and it should not be a problem that it is a form control, just as any other button is a form controll. Because as long as you do not give it a `name` attribute it will not be part of your Form Data. (Also I do not think many uses Form Data any longer to construct a data-set.

Quote from the interweb: "only fields (“controls”) that have a name attribute contribute to the form data set."